website/server/models/user/methods.js

```diff
@@ -150,11 +150,17 @@ schema.methods.sendMessage = async function sendMessage (userToReceiveMessage, o
  * @param  seen  If the notification should be marked as seen
  */
 schema.methods.addNotification = function addUserNotification (type, data = {}, seen = false) {
-  this.notifications.push({
-    type,
-    data,
-    seen,
-  });
+  // only check if chat notification exists
+  const notifactionAlreadyExists = type === 'NEW_CHAT_MESSAGE' ?
```

Comment:
I don't think this will make any difference because chat notifications are added directly to the db using a different method https://github.com/HabitRPG/habitica/blob/develop/website/server/models/group.js#L603

Reply:
oh wow o.O this place I didn't even saw at all ..., well gonna try to fix it there

Reply:
is there a way to have a check before `$push` is called?

Reply:
what type of check? You can make another call to the db but won't change anything as it's without transactions anyway

Reply:
for this: `User.update(query, lastSeenUpdateAddNew, {multi: true}).exec();` 

I'd like to add a check if this notification already added, or would it be ok to go through each user and do the check+ add of notification instead of the

Reply:
Not that I know and tbh it's not something so serious to require an extra query imho, also because if you add a second query you can still have the same problems as now